protos/flyteidl/datacatalog/datacatalog.proto

```diff
@@ -150,6 +158,39 @@ message ListDatasetsResponse {
     string next_token = 2;
 }
 
+// Get the Artifact or try to reserve a spot if the Artifact does not exist.
+message GetOrReserveArtifactRequest {
```

Comment:
Should we have sort of an owner_id? so that if the same owner keeps showing up to reserve, it can respond with Acquired... and refresh the timeout

Reply:
So the clock starts when a task successfully "reserve" a spot, the task then continue with the execution. If the task does not finish within the timeout, other tasks can take the reservation and kick off another execution.